app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -17,42 +17,65 @@ package com.keylesspalace.tusky.db
 
 import android.content.SharedPreferences
 import android.util.Log
+import androidx.room.withTransaction
 import com.keylesspalace.tusky.db.dao.AccountDao
 import com.keylesspalace.tusky.db.entity.AccountEntity
+import com.keylesspalace.tusky.di.ApplicationScope
 import com.keylesspalace.tusky.entity.Account
 import com.keylesspalace.tusky.entity.Status
 import com.keylesspalace.tusky.settings.PrefKeys
 import java.util.Locale
 import javax.inject.Inject
 import javax.inject.Singleton
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.runBlocking
 
 /**
- * This class caches the account database and handles all account related operations
- * @author ConnyDuck
+ * This class is the main interface to all account related operations.
  */
 
 private const val TAG = "AccountManager"
 
 @Singleton
 class AccountManager @Inject constructor(
-    db: AppDatabase,
-    private val preferences: SharedPreferences
+    private val db: AppDatabase,
+    private val preferences: SharedPreferences,
+    @ApplicationScope private val applicationScope: CoroutineScope
 ) {
 
-    @Volatile
-    var activeAccount: AccountEntity? = null
-        private set
+    private val accountDao: AccountDao = db.accountDao()
 
-    var accounts: MutableList<AccountEntity> = mutableListOf()
-        private set
+    /** A StateFlow that will update everytime an account in the database changes, is added or removed.
+     *  The first account is the currently active one.
+     */
+    val accountsFlow: StateFlow<List<AccountEntity>> = runBlocking {
+        accountDao.allAccounts()
+            .stateIn(CoroutineScope(applicationScope.coroutineContext + Dispatchers.IO))
+    }
```

Comment:
We do a lot of searching the list by id, would it be nontrivial to make this a map of id to account?

Reply:
Yes that should be easy with [`associate`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/associate.html). But actually the "a lot of searching the list by id" is a mistake, this should only happen once in AccountManager and everything else should use that code, let me change this.